Explorer/Assets/DCL/PerformanceAndDiagnostics/Optimization/AdaptivePerformance/Systems/AdaptivePerformancePlugin.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Arch.Core;
+using Arch.SystemGroups;
+using Cysharp.Threading.Tasks;
+using DCL.AssetsProvision;
+using DCL.PluginSystem.Global;
+using DCL.Profiling;
+using DCL.RealmNavigation;
+using System.Threading;
+using UnityEngine;
+
+namespace DCL.Optimization.AdaptivePerformance.Systems
+{
+    public class AdaptivePerformancePlugin : IDCLGlobalPlugin<AdaptivePerformanceSettings>
+    {
+        private readonly IAssetsProvisioner assetsProvisioner;
+        private readonly Profiler profiler;
+        private readonly ILoadingStatus loadingStatus;
+
+        private ProvidedAsset<AdaptivePhysicsSettings> physicsSettings;
+
+        public AdaptivePerformancePlugin(IAssetsProvisioner assetsProvisioner, Profiler profiler, ILoadingStatus loadingStatus)
+        {
+            this.assetsProvisioner = assetsProvisioner;
+            this.profiler = profiler;
+            this.loadingStatus = loadingStatus;
+        }
+
+        public async UniTask InitializeAsync(AdaptivePerformanceSettings settings, CancellationToken ct)
+        {
+            physicsSettings = await assetsProvisioner.ProvideMainAssetAsync(settings.phyiscsSettings, ct);
+
+            Physics.simulationMode = physicsSettings.Value.Mode == PhysSimulationMode.MANUAL ? SimulationMode.Script : SimulationMode.FixedUpdate;
+        }
+
+        public void InjectToWorld(ref ArchSystemsWorldBuilder<World> builder, in GlobalPluginArguments arguments)
+        {
+            UpdatePhysicsSimulationSystem.InjectToWorld(ref builder, profiler, physicsSettings.Value, loadingStatus);
+            AdaptPhysicsSystem.InjectToWorld(ref builder, profiler, physicsSettings.Value, loadingStatus);
```

Comment:
Does it make sense to inject this when physics update is set to manual?

Reply:
I left this one and others similar just to have possibility to switch it at runtime, in case of debugging some physics-related bugs. We can revise it after some time again and if we all good with the physics behavior, then not inject these systems at all. As for now I would leave runtime switch possibility